crates/common/tedge_utils/src/file.rs

```diff
@@ -58,6 +58,15 @@ pub fn create_directory_with_mode(dir: impl AsRef<Path>, mode: u32) -> Result<()
     perm_entry.create_directory(dir.as_ref())
 }
 
+pub fn create_file_with_mode(
```

Comment:
This can be done independently as this PR is already large.

That said I would suggest yet another direction (for later!). I agree with Albin that we cannot have a method per combination, but I also understand the need for something more explicit than a function with 3 optional arguments. So I would add `with_user`, `with_group` and `with_mode` method to `PermissionEntry`. So we could write: `PermissionEntry::new().with_mode(644).create_file(path, content)`.